spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -174,12 +174,17 @@ public long targetDataFileSize() {
         .parse();
   }
 
-  public boolean fanoutWriterEnabled() {
+  public boolean useFanoutWriter(SparkWriteRequirements writeRequirements) {
+    boolean defaultValue = !writeRequirements.hasOrdering();
+    return useFanoutWriter(defaultValue);
+  }
+
+  private boolean useFanoutWriter(boolean defaultValue) {
```

Comment:
I can simply expose this version and get rid of the one above. However, it felt a bit safer to only expose the one accepting write requirements to send the message the default value depends on how the data is ordered.

Thoughts, @RussellSpitzer @rdblue?

Reply:
Any ideas on how to avoid that? It is a default value in this case.

Reply:
An internal method with a boolean would be fine. Or maybe have a method like `fanoutWriterEnabled()` vs `useFanoutWriter`.

Reply:
This one is already private, let me play around with this locally.